### PR TITLE
Create database and table

### PR DESCRIPTION
**Pull request by SeppeVerhavert**
Create notes_database, called notes_db for quicker referencing, same for notes_tb.
If one already exists it skips creating it.

My title is a VARCHAR with a limit of 30 characters because I wanted to put a limit on the length of my titles. Title also can't be empty, otherwise you can't search on titles.

My id is a SMALLINT, which comes in a smaller format of INT but still has a range of 65000+. This will be more than enough. Id is auto-incremented so it doesn't  need manual input and is generated on creating the note. 

My modifieddate is a TIMESTAMP, which uses automatic initialization and updating to the current date and time using DEFAULT CURRENT_TIMESTAMP and ON UPDATE CURRENT_TIMESTAMP in the column definition.

My note is a VARCHAR(MAX) so notes can be of almost unlimited length. The default of note is NULL so it is possible to create an empty note with just a title.

My PRIMARY KEY will be id, so each note will have a unique value.

### CODE REVIEW

File: noteDB.sql [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CREATE DATABASE IF NOT EXISTS 'notesDB';
```
Comment (johnnycalderon86) [inline noteDB.sql]: Not sure if its good to put the database name in single quotes.

Comment (SeppeVerhavert) [reply noteDB.sql]: Me neither, this is the syntax I found online. I'll look it up

File: noteDB.sql [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+CREATE TABLE IF NOT EXISTS 'notesTB' (
+    title VARCHAR(30) NOT NULL,
+    id SMALLINT NOT NULL AUTO_INCREMENT,
+    modifieddate TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+    note TEXT DEFAULT NULL,
```
Comment (jasmine8711) [inline noteDB.sql]: i think it is good using TEXT on the note, if you want the length of note unlimited.
Is it necessary  to set the NULL on your TEXT col?

File: noteDB.sql (hunk 1 above)
Comment (SeppeVerhavert) [reply noteDB.sql]: According to w3schools you don't even need the single quotes.
https://www.w3schools.com/sql/sql_create_db.asp

```sql
CREATE DATABASE databasename; 
```

File: noteDB.sql (hunk 2 above)
Comment (jasmine8711) [reply noteDB.sql]: why the default is NULL instead ""(empty string)

Comment (SeppeVerhavert) [reply noteDB.sql]: Apparently TEXT is old and should not be applied to new code.
VARCHAR(MAX) should be the better option here, fixed this!
No it is not necessary but I felt you should be able to make a note with only a title and empty text.

**Comment by iliastic** [comment]
modifieddate TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP 
How does it know the time is it from an api or the device itself?

File: noteDB.sql (hunk 2 above)
Comment (SeppeVerhavert) [reply noteDB.sql]: According to Caché Development Guides
https://cedocs.intersystems.com/latest/csp/docbook/DocBook.UI.Page.cls?KEY=GSQL_langelements_null

<q>The SQL zero-length string is not recommended for use as a field input value or a field default value. Use NULL to represent the absence of a data value.</q>

**Comment by SeppeVerhavert** [comment]
> modifieddate TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP
> How does it know the time is it from an api or the device itself?

According to SQLServertutorial.net
http://www.sqlservertutorial.net/sql-server-date-functions/sql-server-current_time-function/

The CURRENT_TIMESTAMP function returns the current timestamp of the operating system of the server on which the SQL Server Database runs. The returned timestamp is a DATETIME value without the time zone offset.

File: noteDB.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+CREATE DATABASE IF NOT EXISTS 'notesDB';
+
+CREATE TABLE IF NOT EXISTS 'notesTB' (
+    title VARCHAR(30) NOT NULL,
+    id SMALLINT NOT NULL AUTO_INCREMENT,
+    modifieddate TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+    note VARCHAR(MAX) DEFAULT NULL,
+    PRIMARY KEY (id)
+)
```
Comment (Ilger) [inline noteDB.sql]: Statments always end with a semicolon `;`

**Review by Ilger — CHANGES_REQUESTED** [review]

File: noteDB.sql (hunk 3 above)
Comment (SeppeVerhavert) [reply noteDB.sql]: Thanks man.

**Pull request merged**